### ISSUE TITLE
[FEATURE] Deprecated function detection rule

### ISSUE THREAD

**Issue by etknorr** [bw-design-group/ignition-lint#87]
## Overview

Add a rule that detects deprecated Ignition scripting functions in inline scripts and suggests modern replacements, filtered by the project's configured Ignition version.

## Problem

Projects that haven't been fully migrated contain deprecated calls that still work today but will break on future upgrades. The linter should catch these early and point to the correct replacement.

## Proposed Rule

**Rule ID:** `deprecated_function` | **Category:** `scripts`

## Deprecations

| Deprecated | Replacement | Since |
|---|---|---|
| `system.tag.read/readAll` | `system.tag.readBlocking` | 8.1 |
| `system.tag.write/writeAll` | `system.tag.writeAsync` | 8.1 |
| `system.tag.writeSynchronous/writeAllSynchronous` | `system.tag.writeBlocking` | 8.1 |
| `system.tag.editTag/addTag` | `system.tag.configure` | 8.1 |
| `system.tag.browseConfiguration` | `system.tag.getConfiguration` | 8.1 |
| `system.tag.queryTagHistory` | `system.historian.queryRawPoints` | 8.3 |
| `system.tag.queryTagCalculations/queryTagDensity` | `system.historian.queryAggregatedPoints` | 8.3 |
| `system.tag.browseHistoricalTags` | `system.historian.browse` | 8.3 |
| `system.tag.storeTagHistory` | `system.historian.storeDataPoints` | 8.3 |
| `system.tag.storeAnnotations/deleteAnnotations/queryAnnotations` | `system.historian.*` | 8.3 |
| `system.db.runQuery` | `system.db.execQuery` / `system.db.runPrepQuery` | 8.3 |
| `system.db.runUpdateQuery` | `system.db.execUpdate` / `system.db.runPrepUpdate` | 8.3 |
| `system.db.runScalarQuery` | `system.db.execScalar` | 8.3 |
| `system.db.runNamedQuery` | `system.db.execQuery` | 8.3 |
| `system.db.runSFUpdateQuery` | `system.db.execUpdateAsync` | 8.3 |
| `system.db.dateFormat` | `system.date.format` | 8.3 |
| `system.db.clearAllNamedQueryCaches/clearNamedQueryCache` | `system.db.clearCache` | 8.3 |
| `system.net.httpGet/Post/Put/Delete` | `system.net.httpClient.*` | 8.3 |
| `system.gui.*` (vision) | `system.vision.*` | 8.3 |
| `system.nav.*` (vision) | `system.vision.*` | 8.3 |
| `system.util.invokeGateway` | `system.util.sendRequest` | 7.9 |
| `system.alert.*` | `system.alarm.*` | 7.6 |

## Configuration

```json
{
  "ignition_version": "8.3",
  "rules": {
    "deprecated_function": {
      "severity": "warning",
      "ignored_functions": []
    }
  }
}
```
